src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -152,5 +152,134 @@ public static void GetHRForException()
                 Marshal.GetExceptionForHR(e.HResult);
             }
         }
+
+        [Fact]
+        public static void GenerateGuidForType()
+        {
+             Assert.Throws<PlatformNotSupportedException>(() => Marshal.GenerateGuidForType(typeof(System.Int32)));        
+        }
+
+        [Fact]
+        public static void GenerateProgIdForType()
+        {
+             Assert.Throws<PlatformNotSupportedException>(() => Marshal.GenerateProgIdForType(typeof(System.Int32)));        
```

Comment:
All of these PNSE-verifying tests are going to fail when run on desktop, right?  If so, they should be conditioned to not run on NetFramework.